app/vmselect/promql/rollup_result_cache.go

```diff
@@ -442,8 +442,7 @@ func mergeTimeseries(a, b []*timeseries, bStart int64, ec *EvalConfig) []*timese
 	defer bbPool.Put(bb)
 	for _, ts := range a {
 		bb.B = marshalMetricNameSorted(bb.B[:0], &ts.MetricName)
-		k := bytesutil.InternBytes(bb.B)
-		m[k] = ts
+		m[string(bb.B)] = ts
```

Comment:
No, it cannot (at least with go 1.20-21 versions).

 Makes copy of []byte: `m[string(bb.B)] = ts` 
 Maybe unsafe convert []byte to string: `fmt.Println(m[string(bb.B)])` with `slicebytetostringtmp` runtime function.

Reply:
Based on https://go.dev/src/runtime/string.go
```
// slicebytetostringtmp returns a "string" referring to the actual []byte bytes.
//
// Callers need to ensure that the returned string will not be used after
// the calling goroutine modifies the original slice or synchronizes with
// another goroutine.
// Some internal compiler optimizations use this function.
//   - Used for m[T1{... Tn{..., string(k), ...} ...}] and m[string(k)]
```

`m[string(k)]` looks like exactly what we do here. And after that we reuse `bb.B` again.

Do I understand it right that key won't be corrupted because we're still in the scope of the same goroutine here?

Reply:
Previous comment was more clear for this function:

```
// First such case is a m[string(k)] lookup where
	// m is a string-keyed map and k is a []byte.
	// Second such case is "<"+string(b)+">" concatenation where b is []byte.
	// Third such case is string(b)=="foo" comparison where b is []byte.
```

 Only map key lookup uses given optimization.